clients/src/main/java/org/apache/kafka/common/metrics/KafkaMetric.java

```diff
@@ -40,15 +48,30 @@ public KafkaMetric(Object lock, MetricName metricName, MetricValueProvider<?> va
         this.time = time;
     }
 
+    /**
+     * Get the configuration of this metric
+     * @return Return the config of this metric
+     */
     public MetricConfig config() {
-        return this.config;
+        synchronized (lock) {
```

Comment:
Could you please file another jira to trace it? This fix is good but it should be in another PR.

Reply:
sure, let me address this. I think you are right, this is a different issue. Thanks for pointing this out!

Reply:
this is addressed in [#15550](https://github.com/apache/kafka/pull/15550)